src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs

```diff
@@ -755,6 +757,298 @@ public void IncludeWithWildcardShouldNotPreserveUserSlashesInFixedDirectoryPart(
+                {
+                    ChangeWaves.ResetStateForTests();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Throws exception when an imported wildcard evaluates to a drive enumeration.
+        /// </summary>
+        [Fact]
+        public void ImportEvaluationResultsInDriveEnumerationException()
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                ChangeWaves.ResetStateForTests();
+                env.SetEnvironmentVariable("MSBUILDFAILONDRIVEENUMERATINGWILDCARD", "1");
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
+                try
+                {
+                    string content =
+                    @"<Project xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+                        <Import Project=""$(Microsoft_WindowsAzure_EngSys)\**\*"" />
+                    </Project>
+                    ";
+
+                    var testFile = env.CreateFile(env.CreateFolder(), "a.csproj", content);
+                    Should.Throw<InvalidProjectFileException>(() => { ProjectInstance.FromFile(testFile.Path, new ProjectOptions()); });
+                }
+                finally
+                {
+                    ChangeWaves.ResetStateForTests();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Logs warning when an imported wildcard evaluates to a drive enumeration.
+        /// </summary>
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/msbuild/issues/7330")]
+        public void ImportEvaluationResultsInDriveEnumerationWarning()
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                ChangeWaves.ResetStateForTests();
+                env.SetEnvironmentVariable("MSBUILDFAILONDRIVEENUMERATINGWILDCARD", "0");
+                BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
+
+                try
+                {
+                    string content =
+                    @"<Project xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+                          <Import Project=""z:\**\*.targets"" />
+                      </Project>
+                    ";
+
+                    string tempPath = Path.GetTempPath();
+                    var testFile = env.CreateFile(Path.Combine(tempPath, "a.csproj"), content);
+                    ProjectOptions options = new ProjectOptions();
+                    options.ProjectCollection = new ProjectCollection();
+
+                    MockLogger collectionLogger = new MockLogger();
+                    options.ProjectCollection.RegisterLogger(collectionLogger);
+
+                    ProjectInstance.FromFile(testFile.Path, options);
+
+                    collectionLogger.WarningCount.ShouldBe(1);
+                    collectionLogger.AssertLogContains("MSB5029");
+                    options.ProjectCollection.UnregisterAllLoggers();
```

Comment:
Should I be using `[Theory]'s` for this as well instead of introducing a helper? Or do a combination of both for these scenarios since it is also duplicated in `ProjectItemInstance_Tests.cs`?

Reply:
Ideally a combination as it's orthogonal (removing code duplication vs. a more complete test matrix) but seeing how well `FileMatcher` is covered with new tests I would prioritize introducing helpers.